specs/agents/tracing-instrumentation-aws-lambda.md

```diff
@@ -0,0 +1,158 @@
+# AWS Lambda instrumentation
+This spec specifies the instrumentation of applications / services running on AWS Lambda.
+
+An AWS Lambda application needs to implement a **handler method** that is called whenever that Lambda function is invoked. A handler method receives (at least) two objects with useful information:
+- `event`: Depending on the trigger type of the Lambda function this object contains different, trigger-specific data. For some trigger types this object may be empty/null.
+- `context`: This object provides generic (trigger agnostic) meta information about the Lambda function.
+
+In our instrumentation we use these objects to derive useful meta and context information.
+
+## Generic Lambda Instrumentation
+In general, to instrument Lambda functions, we create transactions that wrap the execution of that handler method. In cases where we cannot assume any type or information in the `event` object (e.g. if the trigger type is undefined), we ignore trigger-specific information and simply wrap the handler method with a transaction, while using the `context` object to derive some necessary fields.
+
+Field | Value | Description 
+---   | ---   | ---
+`transaction.name` | e.g. `MyFunctionName` | The name of the Lambda function. This can be retrieved either from the `context` object or from the environment variable `AWS_LAMBDA_FUNCTION_NAME`
+`transaction.type` | `request` |  
+`faas.id` | e.g. `arn:aws:lambda:us-west-2:123456789012:function:my-function` | Use `context.invokedFunctionArn` to set the id. 
```

Comment:
Great catch!

Regarding first comment: I'll add a note to remove the alias suffix when using `context.invokedFunctionArn`. 

To the second quote: Yes, the account ID is not available at start-up of the environment, that's why we need to set this as a transaction context field instead of a metadata field. IIRC, the `context.invokedFunctionArn` should contain the correct account ID